gaia-fast-list.js

```diff
@@ -204,29 +252,39 @@ Internal.prototype = {
   itemHeight: 60,
 
   setModel: function(model) {
+    debug('set model', model);
     if (!model) return;
     this.model = model;
     this.sections = this.sectionize(model);
     if (!this.fastList) this.createList();
     else this.fastList.reloadData();
+    this.setEmpty(false);
+  },
+
+  setEmpty: function(value) {
+    this.container.classList.toggle('empty', value);
   },
 
   sectionize: function(items) {
     var hash = {};
+
     if (!this.getSectionName) { return hash; }
 
-    items.forEach(item => {
-      var section = this.getSectionName(item);
+    for (var i = 0, l = items.length; i < l; i++) {
+      var section = this.getSectionName(items[i]);
       if (!section) { return; }
       if (!hash[section]) { hash[section] = []; }
-      hash[section].push(item);
-    });
+      hash[section].push(items[i]);
+    }
 
     return hash;
   },
 
   createList: function() {
+    debug('create list');
     this.fastList = new FastList(this);
```

Comment:
good point, maybe just `updateCache` then (can be done inside the method, so it returns a scheduler promise)